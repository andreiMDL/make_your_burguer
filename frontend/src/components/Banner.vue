<template>
  <div class="main-banner">
    <!-- Adicionar animação de transição -->
    <a @click="scrollToForm" href="#div-pedido">Make Your Burger</a>
    
  </div>
</template>

<script>
export default {
  name: "Banner",
  methods: {

    scrollToForm(event) {
      event.preventDefault();
      const section = document.querySelector(".burger-form");
      if (section) {
        section.scrollIntoView({ behavior: "smooth" });
      }
    }
  }
}
</script>

<style scoped>
.main-banner {
  background-image: url("/public/img/burger.jpg");
  background-position: center;
  background-size: cover;
  min-height: 600px;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 4px solid #080808;
  box-shadow: 0px 0px 25px rgba(0, 0, 0, 0.761);
}

.main-banner a {
  
  text-decoration: none; 
  margin-left: 6%;
  border-radius: 25px;
  font-family: "Marck Script", serif; 
  font-size: 350%;
  border: 1px solid #ff9100;
  margin-left: 8% ;
  color: #ff9100;
  text-align: center;
  background-color: #08060000;
  padding: 1% 3%;
  transition: .5s;
  text-shadow: 0px 0px 30px rgba(255, 166, 0, 0.5);

}
.main-banner a:hover {
  font-weight: bold;
  box-shadow: 0px 0px 20px rgba(255, 140, 0, 0.5);
  text-shadow: 0px 0px 20px rgba(112, 62, 1, 0.5);
  color: #9e5a00;
  
  background-color: #cab989;
  
}


@media (max-width: 1061px) {
  .main-banner {
    background-position: right bottom;
    }
  }
</style>