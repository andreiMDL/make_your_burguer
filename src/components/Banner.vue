<template>
  <div id="main-banner">
    <h1>Make Your burger</h1>

  </div>
</template>

<script>
export default {
  name: "Banner"
}
</script>

<style scoped>
#main-banner {
  background-image: url("/public/img/burger.jpg");
  background-position: 100%;
  background-size: cover;
  height: 96vh;
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

#main-banner h1 {
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  border: 3px solid #fcba03;
  border-left: 2px solid #222;
  margin-top: -100px;
  color: #fff;
  text-align: center;
  font-size: 60px;
  background-color: #222;
  padding: 20px 40px;
}

@media (max-width: 1061px) {
  #main-banner {
    background-position: right bottom;
    }
  }
</style>