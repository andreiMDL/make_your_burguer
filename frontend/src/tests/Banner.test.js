import { expect, it, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import Banner from '../components/Banner.vue'


// Renderiza o Banner
it('should render the button with the correct text', () => {
    const wrapper = mount(Banner)

    //Verifica se o botão existe no html renderizado
    const button = wrapper.find('a')

    //Verifica se o botão está no DOM
    expect(button.exists()).toBe(true)

    //Verifica se o texto do botão está correto
    expect(button.text()).toBe('Make Your Burger')

    

})

it('should have the correct attributes on the button', () => {
    const wrapper = mount(Banner)
    const button = wrapper.find('a')

    expect(button.attributes('href')).toBe('#div-pedido')
})

it('should call scrollToForm when the button is clicked', async () => {
    const wrapper = mount(Banner)
    const spy = vi.spyOn(wrapper.vm, 'scrollToForm') // Espiona diretamente a função do wrapper

    const button = wrapper.find('a')
    await button.trigger('click')

    expect(spy).toHaveBeenCalledOnce()
})

it('should prevent default event on button click', async () => {
    const wrapper = mount(Banner)
    const button = wrapper.find('a')

    const event = { preventDefault: vi.fn() }
    await wrapper.vm.scrollToForm(event) // Chama diretamente a função

    expect(event.preventDefault).toHaveBeenCalled()
})

it(' should have the correct CSS class applied', () => {
    const wrapper = mount(Banner)

    expect(wrapper.find('.main-banner').exists()).toBe(true)
    //Certifica-se que a classe existe
})


it('should have a valid href attribute', () => {
    const wrapper = mount(Banner)
    const button = wrapper.find('a')

    expect(button.attributes('href')).toBe('#div-pedido')
})

it('should prevent default behavior when button is clicked', async () => {
    const wrapper = mount(Banner)
    const button = wrapper.find('a')

    const event = { preventDefault: vi.fn() }
    await button.trigger('click', event)

    expect(event.preventDefault).toHaveBeenCalledOnce()
})

it('should match the snapshot', () => {
    const wrapper = mount(Banner)

    expect(wrapper.find('.main-banner').exists()).toBe(true)
})

