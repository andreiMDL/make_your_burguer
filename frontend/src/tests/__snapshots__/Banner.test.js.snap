// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
"<div data-v-3587acbb="" id="main-banner">
  <!-- Adicionar animação de transição --><a data-v-3587acbb="" href="#div-pedido">Make Your Burger</a>
</div>"
`;
